### ISSUE TITLE
Use XJC/JAXB to generate Java classes of SGr specification

### ISSUE THREAD

**Issue by mkrebs81** [SmartGridready/SGrJava#89]
Using Eclipse/EMF to generate classes is quite tedious.

Using XJC can be much easier, because it has a Gradle plugin (https://github.com/bjornvester/xjc-gradle-plugin) and does not need an installation of the Eclipse IDE.

The downside is that all code which accesses the EMF classes needs to be adapted.

### Tasks

- [x] workaround for singleton factory
- [x] alternative solution for V0Package constants
- [x] alternative for EStructuralFeature and dynamic type names
- [x] alternative for complex types with choices, e.g. RestApiDataPointConfiguration --> RestApiServiceCall
- [x] adapt EID loader

**Comment by mkrebs81** [comment]
### Advantages

- class generation is **MUCH** easier than using Eclipse EMF

### Disadvantages

- no singleton _V0Factory_, _ObjectFactory_ instead
- no _V0Package_ with data type constants
- XML loader needs modification

**Comment by mkrebs81** [comment]
it seems that XJC does not like enum types that have values starting with a digit...

found this workaround: https://stackoverflow.com/questions/30370456/generating-big-enum-classes-with-numbers-in-enum-values-jaxb-xjc

it works, but the generated names are `VALUE_1`, `VALUE_2` etc.

**Comment by mkrebs81** [comment]
`RestApiDataPointConfiguration` is not generated correctly --> properties reference to JAXB elements instead of the correct classes

**Comment by mkrebs81** [comment]
the whole `EStructualFeature` stuff does not exist in JAXB. getting type names dynamically may get complicated.

**Comment by ergo-furrer** [comment]
The code using JAXB works now.
However I found that there is a dependency to in the sgr-driver-api that uses generated classes (MQTT-interface). I think I will remove that. 
To come over conflicts with the EMF i did temporarily publish JAXB versions of the libraries e.g.  1.1.0-JAXB-SNAPSHOT ....

**Comment by mkrebs81** [comment]
Thank you. I did not have time to look at it last week, but I added some modifications today:
1. parametrization of device descriptions, so test do not fail
2. added JAXB-SNAPSHOT to SGrJava as well, so I can distinguish versions
3. added an issue and branch to SGrJavaDrivers, so I can compile with JAXB and publish to local repo

I have noticed one issue with the JAXB implementation:
The current "0.3.0" version of the ABB B23 EID cannot be loaded - unfortunately I have not yet found the issue; the exception is unclear...
